discovery/azure/azure.go

```diff
@@ -317,3 +345,115 @@ func (d *Discovery) refresh() (tg *targetgroup.Group, err error) {
 
 	return tg, nil
 }
+
+func (client *azureClient) getVMs() (vms []virtualMachine, err error) {
```

Comment:
Can you avoid named return values like this? It's harder to understand the code.

Reply:
I changed this for the other methods as well.